model/src/lib.rs

```diff
@@ -51,35 +52,94 @@ pub struct UserOrder {
     pub signature: Signature,
 }
 
-impl UserOrder {
+impl OrderCreation {
     pub fn token_pair(&self) -> Option<TokenPair> {
         TokenPair::new(self.buy_token, self.sell_token)
     }
 }
 
-/// An order that is returned when querying the orderbook.
-///
-/// Contains extra fields thats are populated by the orderbook.
-#[derive(Eq, PartialEq, Clone, Copy, Debug, Deserialize, Serialize)]
+// uid as 56 bytes: 32 for orderDigest, 20 for ownerAddress and 4 for validTo
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct OrderUid(pub [u8; 56]);
+
+impl Default for OrderUid {
+    fn default() -> Self {
+        OrderUid([0 as u8; 56])
+    }
+}
+
+impl Serialize for OrderUid {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        // Todo: build in underscores
```

Comment:
Maybe it makes sense not to use underscores in the end so the UID can be pasted into Etherscan. WDYT? Maybe prefix with `0x`?

Reply:
According to the openapi spec there are no underscores in the string, just hex encoded bytes without 0x. No 0x is also what I specified for other fields in the openapi spec.

Reply:
Okay, let's keep the underscores out. 
I see that `0x` prefix could be convenient, though we don't use them for any API interface, and hence it would be little bit weird. I guess users can add 0x easily themselves. Hopefully not too many users will do that anyways.